submissions/Serge044/Classic-Frogger-Game/app.js

```diff
@@ -0,0 +1,151 @@
+const Enemy = function (x, y) {
+  this.width = 75;
+  this.x = x;
+  this.y = y;
+  this.border = { left: -100, right: 505 };
+  this.speed = getRandomNum(200, 300);
+  this.sprite = "images/tank-icon.png";
+};
+
+Enemy.prototype.update = function (dt) {
+  if (this.x < this.border.right) {
+    this.x += dt * this.speed + Math.random();
+  } else {
+    this.speed = getRandomNum(200, 300);
+    this.x = this.border.left;
+  }
+};
+
+function getRandomNum(min, max) {
+  return Math.floor(Math.random() * (max - min + 1)) + min;
+}
+
+Enemy.prototype.render = function () {
+  ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
+};
+
+const Player = function () {
+  this.width = 75;
+  this.x = 202;
+  this.y = 400;
+  this.stepX = 101;
+  this.stepY = 84;
+  this.border = { top: -20, right: 404, bottom: 400, left: 0 };
+  this.sprite = "images/char-nafo.png";
+};
+
+Player.prototype.render = function () {
+  ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
+};
+
+Player.prototype.toStartPosition = function () {
+  this.x = 202;
+  this.y = 400;
+};
+
+Player.prototype.update = function () {
+  if (this.y === this.border.top) {
+    setTimeout(() => {
+      swal(
+        "Congratulations!",
+        "Crimea finally under the Ukrainian flag 🇺🇦",
+        "success",
+        {
+          button: "Слава Україні!",
+        }
+      );
+      this.toStartPosition();
+    }, 100);
+  }
+  this.checkEnemyCollision();
+};
+
+Player.prototype.checkEnemyCollision = function () {
+  allEnemies.forEach((el) => {
```

Comment:
See item 4.iii of the [requirements](https://kottans.org/documentation/docs/doc/code-review/#oo-js-frogger).

Also naming. `el` is too generic and is an abbreviation. Code is written for other people to read.